src/batch_extractor.cc

```diff
@@ -66,7 +67,8 @@ rocksdb::Status WriteBatchExtractor::PutCF(uint32_t column_family_id, const Slic
     } else if (metadata.expire > 0) {
       auto args = log_data_.GetArguments();
       if (args->size() > 0) {
-        RedisCommand cmd = static_cast<RedisCommand >(std::stoi((*args)[0]));
+        auto parse_result = ParseInt<int>((*args)[0], 10);
+        RedisCommand cmd = static_cast<RedisCommand >(*parse_result);
```

Comment:
If you do not check whether the `parse_result` is ok before getting its value, it will abort the program when facing an error status in `ParseInt`.

Reply:
> If you do not check whether the `parse_result` is ok before getting its value, it will abort the program when facing an error status in `ParseInt`.

Hi. The previous code also didn't the correctness of parsing. My understanding is that this is a certain successful parsing. Do you think we need to do extra checks?

Reply:
The behavior is not quite same as previous. While using `sto*` or `atoi`, the program does not abort if facing some error input.

Reply:
> The behavior is not quite same as previous. While using `ato*`, the program does not abort if facing some error input and do not check result. For `sto*`, it will just throw the exception to the upper layer.

Yes. But if the parsing is expected must to be successful, core dump may be a good choice. Adding defensive check is also a good solution. I'll add the check later.

Reply:
> > The behavior is not quite same as previous. While using `ato*`, the program does not abort if facing some error input and do not check result. For `sto*`, it will just throw the exception to the upper layer.
> 
> Yes. But if the parsing is expected must to be successful, core dump may be a good choice. Adding defensive check is also a good solution. I'll add the check later.

If the parsing is really expected to be successful, you can add some comment to simply prove it (i.e. the condition "result is not ok" is always unreachable whenever the execution path varies). Then you can omit the status checking.